docs/proposals/clusters.md

```diff
@@ -0,0 +1,214 @@
+interfaces:
+  - type: physical
+    subnet: vagrant1
+    mac: 00:00:00:00:00:03
+    ipv4: 127.0.0.3
+metadata:
+    mstr: etcd3
+    random: data
+workflow: something-else
+```
+
+To accomodate logically separated pools of hosts, multiple inventory files will be specified, with each inventory file providing the hosts that belong to a single cluster in the same format shown above:
+
+```
+├── sites
+│   └── site-name
+│       ├── env.yml <- Specifies agent configurations, site-wide metadata, subnet information
+│       ├── cluster-one.yml <- All machines associated with cluster-one
+│       └── cluster-two.yml <- All machines associated with cluster-two 
+```
+
+Additionally, each cluster inventory file should be proceeded with a document containing metadata specific to that collection of machines:
+
+```
+---
+# Cluster metadata
+metadata:
+  mstr: etcd-mstr
+  http-proxy: 127.0.0.10
+---
+# Host 1
+---
+# Host 2
+---
+# etc...
+```
+
+The metadata here should be merged with site (env) metadata, with cluster metadata taking precedence in the event of a key collision. In this way, env metadata serves as a global "default" metadata, which may be individually overwritten by cluster metadata.
+
+
+## Work Scope
+
+This update will require changes to the Data Model definitions, parsing process, and state engine logic. All of these currently assume all inventory will be collected in one homogenous group.
+
+## Estimated Development Time (SWAG)
+
+Medium complexity. Approx. one month with two developers for implementation, testing, and delivery.
+
+
+
+# Separated Configuration/Inventory
+
+After hosts have been separated into clusters, we consider the process for separating the inventory from the defined configurations. The goal is to contain all inventory, cluster, and site information in one location, and maintain workflow and associated configurations in a separate location.
+
+## Requirements
+
+- Provide per-cluster info for linking to remote configuration
+  - Github configuration for pulling from a remote repository
+  - Local directory configuration for referencing on-disk configs
+- Clusters contain metadata that augments site metadata
+
+## Design
+
+The current logical layout for a data model is:
+
+```
+├── assets
+│   └── Config files, templates
+├── boot
+│   └── Pair configs + OSes, and metadata
+├── os
+│   └── OS kernel, initrd, cmdline information
+├── sites
+│   └── site-name
+│       ├── env.yml <- Specifies agent configurations, site-wide metadata, subnet information
+│       └── inventory.yml <- Host definitions deliminated as individual YAML documents 
+└── workflows
+    └── Chain multiple boots together, inter-workflow dependencies and rollout policy
+```
+
+In the current layout, all site inventory and metadata is stored under `sites`. Everything else is configuration.
+
+
+Separating the data model will place all site inventory and metadata into one location, and all configuration into another:
+
+`Location 1: Inventory`
```

Comment:
Multiple inventory locations?

It may be worthwhile to have a larger conversation re: where do we need to consider the possibility of offering some sort of multi-user support in the software. For example: if we need to support sites / clusters with discreet ownership, how does that impact the structure of the data model, assuming that we want to support GitHub's built-in access control mechanisms.